lib/repl.js

```diff
@@ -772,7 +773,7 @@ function complete(line, callback) {
         name = files[f];
         ext = path.extname(name);
         base = name.slice(0, -ext.length);
-        if (base.match(/-\d+\.\d+(\.\d+)?/) || name === '.npm') {
+        if (versionedFileNamesRe.test(base) || name === '.npm') {
```

Comment:
I looked into this a bit further and this all about the npm cache itself. It is meant to prevent listing all the version folders found in the cache folder (e.g. 1.0.0 in `root/.npm/foo/1.0.0/package`). As far as I see it this does not check the cache folder anymore though and the windows cache folder (`npm-cache` instead of `.npm`) is not excluded either. So I think this part could be safely removed and it's a [legacy](https://github.com/nodejs/node/commit/5c1ffa165f67a2bb0fc726e4b7402249b7a895ab) remnant. I did not try it out though.